### ISSUE TITLE
Different file/folder ownership if install is run through UI

### ISSUE THREAD

**Issue by Graloth** [geerlingguy/drupal-vm#1512]
### Issue Type

 - Bug Report

### Your Environment

```
Vagrant 1.9.5
bash: vboxmanage: command not found
VirtualBox
bash: ansible: command not found
```

### Your OS

 - **Windows** (10, build 15063.483, version 1703)

### Full console output

result of `ll` inside the `/sites` directory.
```
drwxrwxr-x. 3 vagrant vagrant  115 Aug 11 09:04 default
-rw-rw-r--. 1 vagrant vagrant  310 Aug 11 09:04 development.services.yml
-rw-rw-r--. 1 vagrant vagrant 4022 Aug 11 09:04 example.settings.local.php
-rw-rw-r--. 1 vagrant vagrant 2337 Aug 11 09:04 example.sites.php
drwxrwxrwx. 4 apache  apache    79 Aug 11 09:11 installtester.dev
-rw-r--r--. 1 apache  apache   131 Aug 11 09:09 sites.local.php
-rw-r--r--. 1 apache  apache  2624 Aug 11 09:09 sites.php
```

### Summary

While working on getting my custom installation profile to work when run via `drush site-install`, I stumbled upon something that didn't quite make sense to me.

I am running it on a clean Drupal-VM, which only includes the deploy config to a git repo that has my custom installation profile, along with a composer.json so that it can download the modules that my installation profile activates.

After `vagrant up` has run `composer install`, I manually run `drush site-install` with the proper parameters for it have db access etc. (the usual arguments that you can also pass in drupal-vm's config).

My installation profile creates a couple of folders and files within the `/sites` directory, and if it is run via `drush site-install` these have the correct ownership set to `vagrant:vagrant`.

**However** If I run the install via the UI, the files and folders are created with the ownership `apache:apache`, which then causes permission issues, as drupal now runs under `vagrant:vagrant`, but tries to create files or folders inside folders owned by `apache:apache`.

I am able to replicate this behavior thusly:

- Create a custom installation profile that creates a folder inside the `/sites` directory.
- Create a simple composer project that will download drupal along with your installation profile via git.
- Configure a clean Drupal-vm to deploy from a git repository with the composer project.
- Make sure it does not install drupal for you, set `drupal_site_install` to `false`
- Run `vagrant up`
- Once it completes, run the site install via the web UI.
- Check the file/folder ownership of the folder created inside the `/sites` directory.

#### TL;DR
If site is installed via web UI, the folders and files created by the custom installation profile is owned by `apache:apache`, causing permission issues. If installed via `drush site-install` ownership is the correct `vagrant:vagrant`.

Is this intentional or some additional config change I need to make?

**Comment by Graloth** [comment]
Some further testing shows that it only sets ownership to `apache:apache` if the drupal-vm has the following config options set to disable synced folders:

```
vagrant_synced_folders: []
vagrant_synced_folder_default_type: ""
```

If synced folders are enabled, the ownership is the correct `vagrant:vagrant` regardless if you install via UI or drush.  
If synced folders are disabled, the ownership is `apache:apache` only during UI installs, drush installs still use the correct `vagrant:vagrant`.

It seems very confusing to me, why changing the synced folder config would have such an impact on installations via the UI.

**Comment by Graloth** [comment]
I've been able to more accurately replicate the issue by following these steps:

1. Create a custom drupal installation profile that adds a folder inside the `/sites`-directory, using php's `mkdir()` function.
2. Make a composer project that installs drupal and the custom install profile.
3. Clone a fresh copy of drupal-vm to a local directory of your choice and configure it with the following options:
    - Use the `drupal_deploy` config options to deploy the composer project via git.
    - Prevent drupal-vm from installing the drupal site with the config option `drupal_install_site: false`
    - Disable vagrants synced folder via the drupal-vm options:
```
vagrant_synced_folders: []
vagrant_synced_folder_default_type: ""
```
4. Run `vagrant up` and wait for it to finish.
5. Install the drupal site using the web UI.
6. After the step where your custom install creates the folder, check their ownership on the virtual machine.
7. It will now display ownership of the folder as `apache:apache`.

By not including the config options for vagrant synced folders (so that it defaults to the stock drupal-vm config), it will get the correct ownership of `vagrant:vagrant`

**Comment by oxyc** [comment]
Maybe it's the synced folders that's causing the files to be owned by `vagrant:vagrant` even if PHP runs as `apache:apache`.

It does sound like a correct scenario though.

1. When the site is installed with drush it runs using the `vagrant` user causing the `vagrant:vagrant` permission.
2. When the site is installed through the UI, it runs through PHP-FPM which is set to run as [`apache:apache` on RedHat](https://github.com/geerlingguy/ansible-role-php/blob/master/vars/RedHat.yml#L32:L33).

For security purposes in production the webserver should not have write access to folders other than `config/` and `files/`.

**Comment by geerlingguy** [comment]
Yeah, I think this is mostly a 'works as designed' issue—as long as the site works, and Drupal can write files where it needs to, and permissions aren't 777 everywhere, I'm happy.

Is there a scenario where unsecure permissions are added or that breaks the Drupal site functionality? If not, I think we should close this ticket with the understanding that certain generated file permissions can differ depending on how Drupal's installed.

**Comment by Graloth** [comment]
@geerlingguy Agree, working as designed, so I'll close the issue. I haven't found any scenarios for me personally, as any production sites are all deployed via commandline scripts, so permissions there are a non-issue, this has only affected my dev environment.
